lib/jekyll/document.rb

```diff
@@ -103,7 +111,7 @@ def output_ext
     #
     # Returns the basename without the file extname.
     def basename_without_ext
-      @basename_without_ext ||= File.basename(path, '.*')
+      @basename_without_ext ||= File.basename(path, ".*")
```

Comment:
`File.basename(path, File.extname(path))` is much more clear to non-programmers.